api/v1alpha1/timeout_types.go

```diff
@@ -43,12 +43,26 @@ type HTTPTimeout struct {
 }
 
 type ClientTimeout struct {
+	// Timeout settings for TCP.
+	//
+	// +optional
+	TCP *TCPClientTimeout `json:"tcp,omitempty"`
+
 	// Timeout settings for HTTP.
 	//
 	// +optional
 	HTTP *HTTPClientTimeout `json:"http,omitempty"`
 }
 
+type TCPClientTimeout struct {
+	// IdleTimeout for a TCP connection. Idle time is defined as a period in which there are no
```

Comment:
I assume that this will be implemented for the envoy TCP Proxy, right? Maybe we should state that these settings only apply to `tcproute`, `tlsroute` resources, so that users don't expect this timeout to apply also in case of HTTP  (over TCP) connections?

Reply:
Yeah, the state will be necessary to clear confusion. TCPClientTimeout can only be applied on listener whose protocol is TCP or TLS, it'll be used to set Envoy TCP Proxy's idle timeout eventually.